File: frontend_gearguard/lib/models/maintenance_request.dart
/// ===================== STATUS =====================
enum MaintenanceStatus {
  inProgress,
  blocked,
  ready,
}

/// ===================== MODEL =====================
class MaintenanceRequest {
  final String subject;
  final String createdBy;
  final String maintenanceFor;
  final String target;
  final String category;
  final String requestDate;
  final String maintenanceType;
  final String team;
  final String technician;
  final String scheduledDate;
  final String duration;
  final int priority;
  final String company;
  final String notes;
  final String instructions;
  final MaintenanceStatus status;

  MaintenanceRequest({
    required this.subject,
    required this.createdBy,
    required this.maintenanceFor,
    required this.target,
    required this.category,
    required this.requestDate,
    required this.maintenanceType,
    required this.team,
    required this.technician,
    required this.scheduledDate,
    required this.duration,
    required this.priority,
    required this.company,
    required this.notes,
    required this.instructions,
    required this.status,
  });
}

File: backend/src/models/MaintenanceRequest.js
import mongoose from "mongoose";

const maintenanceRequestSchema = new mongoose.Schema(
  {
    companyId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Company",
      required: true,
    },

    createdBy: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },

    equipmentId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Equipment",
      required: true,
    },

    equipmentCategory: {
      type: String, // auto-filled from Equipment
      required: true,
    },

    maintenanceTeamId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "MaintenanceTeam",
      required: true,
    },

    assignedTo: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User", // Technician
      default: null,
    },

    type: {
      type: String,
      enum: ["CORRECTIVE", "PREVENTIVE"],
      required: true,
    },

    priority: {
      type: String,
      enum: ["LOW", "MEDIUM", "HIGH"],
      default: "MEDIUM",
    },

    status: {
      type: String,
      enum: ["NEW", "IN_PROGRESS", "REPAIRED", "SCRAP"],
      default: "NEW",
    },

    scheduledDate: {
      type: Date, // required only for PREVENTIVE
    },

    duration: {
      type: Number, // hours
      default: 0,
    },

    startedAt: Date,
    completedAt: Date,

    isOverdue: {
      type: Boolean,
      default: false,
    },
  },
  { timestamps: true }
);

export default mongoose.model(
  "MaintenanceRequest",
  maintenanceRequestSchema
);
